### PR TITLE
ADM-1009 [frontend]: fix the 4xx error

### PR DESCRIPTION
**Pull request by zhou-yinyuan**
## Summary

...

## Before

_Description_

**Screenshots**
If applicable, add screenshots to help explain behavior of your code.

## After

_Description_

**Screenshots**
If applicable, add screenshots to help explain behavior of your code.

## Note

_Null_

### CODE REVIEW

**Review by davidwangcn9 — CHANGES_REQUESTED** [review]

File: frontend/src/hooks/useGetMetricsStepsEffect.ts [hunk 1]
```diff
@@ -88,7 +88,8 @@ export const useGetMetricsStepsEffect = (): useGetMetricsStepsEffectInterface =>
       setStepFailedStatus(MetricsDataFailStatus.NotFailed);
     } else if (hasRejected && hasFulfilled) {
       const rejectedStep = allStepsRes.find((stepInfo) => stepInfo.status === REJECTED);
-      if ((rejectedStep as PromiseRejectedResult).reason.code == 400) {
+      const code = (rejectedStep as PromiseRejectedResult).reason.code;
+      if (code === 400 || code === 401 || code === 403 || code === 404) {
```
[the remark this answers is not in the document]
Comment (zhou-yinyuan) [reply frontend/src/hooks/useGetMetricsStepsEffect.ts]: maybe no, becase the code may be of string type. `type ResponseCodeType = number | string | undefined | null;`

File: frontend/src/hooks/useGetSourceControlConfigurationBranchEffect.tsx [hunk 2]
```diff
@@ -78,7 +78,7 @@ export const useGetSourceControlConfigurationBranchEffect = (): IUseGetSourceCon
             }),
           ),
         );
-        if (response.code === 400) {
+        if (response.code === 400 || response.code === 401 || response.code === 403 || response.code === 404) {
```
[the remark this answers is not in the document]
Comment (zhou-yinyuan) [reply frontend/src/hooks/useGetSourceControlConfigurationBranchEffect.tsx]: same as above

File: frontend/src/hooks/useGetSourceControlConfigurationCrewEffect.tsx [hunk 3]
```diff
@@ -79,7 +79,8 @@ export const useGetSourceControlConfigurationCrewEffect = (): IUseGetSourceContr
       setStepFailedStatus(MetricsDataFailStatus.NotFailed);
     } else if (hasRejected && hasFulfilled) {
       const rejectedStep = allCrewsRes.find((crewInfo) => crewInfo.status === REJECTED);
-      if ((rejectedStep as PromiseRejectedResult).reason.code == 400) {
+      const code = (rejectedStep as PromiseRejectedResult).reason.code;
+      if (code === 400 || code === 401 || code === 403 || code === 404) {
```
[the remark this answers is not in the document]
Comment (zhou-yinyuan) [reply frontend/src/hooks/useGetSourceControlConfigurationCrewEffect.tsx]: same as above

File: frontend/src/hooks/useGetSourceControlConfigurationRepoEffect.tsx [hunk 4]
```diff
@@ -74,7 +74,8 @@ export const useGetSourceControlConfigurationRepoEffect = (): IUseGetSourceContr
       setStepFailedStatus(MetricsDataFailStatus.NotFailed);
     } else if (hasRejected && hasFulfilled) {
       const rejectedStep = allRepoRes.find((repoInfo) => repoInfo.status === REJECTED);
-      if ((rejectedStep as PromiseRejectedResult).reason.code == 400) {
+      const code = (rejectedStep as PromiseRejectedResult).reason.code;
+      if (code === 400 || code === 401 || code === 403 || code === 404) {
```
[the remark this answers is not in the document]
Comment (zhou-yinyuan) [reply frontend/src/hooks/useGetSourceControlConfigurationRepoEffect.tsx]: same as above

File: frontend/src/hooks/useGetMetricsStepsEffect.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (zhou-yinyuan) [reply frontend/src/hooks/useGetMetricsStepsEffect.ts]: done

File: frontend/src/hooks/useGetSourceControlConfigurationBranchEffect.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (zhou-yinyuan) [reply frontend/src/hooks/useGetSourceControlConfigurationBranchEffect.tsx]: done

File: frontend/src/hooks/useGetSourceControlConfigurationCrewEffect.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (zhou-yinyuan) [reply frontend/src/hooks/useGetSourceControlConfigurationCrewEffect.tsx]: done

File: frontend/src/hooks/useGetSourceControlConfigurationRepoEffect.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (zhou-yinyuan) [reply frontend/src/hooks/useGetSourceControlConfigurationRepoEffect.tsx]: done

File: frontend/src/hooks/useGetSourceControlConfigurationBranchEffect.tsx [hunk 5]
```diff
@@ -78,9 +78,10 @@ export const useGetSourceControlConfigurationBranchEffect = (): IUseGetSourceCon
             }),
           ),
         );
-        if (response.code === 400) {
+        const code: number = response.code as number;
+        if (code >= 400 && code < 500) {
           setStepFailedStatus(MetricsDataFailStatus.AllFailed4xx);
-        } else if (response.code === 200) {
+        } else if (response.code == 200) {
```
Comment (davidwangcn9) [inline frontend/src/hooks/useGetSourceControlConfigurationBranchEffect.tsx:84]: Why change === to ==?

**Review by davidwangcn9 — CHANGES_REQUESTED** [review]

File: frontend/src/hooks/useGetSourceControlConfigurationBranchEffect.tsx (hunk 5 above)
Comment (zhou-yinyuan) [reply frontend/src/hooks/useGetSourceControlConfigurationBranchEffect.tsx:84]: === is ok

**Review by davidwangcn9 — APPROVED** [review]

**Pull request merged**